telegram/src/main/kotlin/com/github/kotlintelegrambot/Bot.kt

```diff
@@ -147,7 +147,10 @@ class Bot private constructor(
             false, null -> {
                 val errorMessage: String = when {
                     error?.message != null -> error.message!!
-                    response?.errorBody() != null -> response.errorBody().toString()
+                    (response?.errorBody()?.contentLength() ?: 0L) != 0L ->
+                        response?.errorBody()!!.string()
+                    response?.raw()?.message() != null ->
+                        "${response.raw().code()} ${response.raw().message()}"
```

Comment:
I was thinking how to make this a bit more readable. What do you think about something like the next?:

```kotlin
            false, null -> {
                fun dispatchableError(message: String) =
                    listOf<TelegramError>(RetrieveUpdatesError(message))
                
                val errorMessage = error?.message
                if (errorMessage != null) {
                    return dispatchableError(errorMessage)
                }

                val errorBody = response?.errorBody()?.string()
                if (errorBody != null && errorBody.isNotEmptyAndNotBlank()) {
                    return dispatchableError(errorBody)
                }

                val rawHttpResponse = response?.raw()
                if (rawHttpResponse != null && rawHttpResponse.message().isNotEmptyAndNotBlank()) {
                    return dispatchableError(
                        "${rawHttpResponse.code()} - ${rawHttpResponse.message()}"
                    )
                }

                return dispatchableError(
                    "There was a problem retrieving updates from Telegram server"
                )
            }
```

Reply:
Yes, in this form, readability is much better.